sdk/python/kfp/dsl/_types.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class BaseType:
+	'''MetaType is a base type for all scalar and artifact types.
+	'''
+	pass
+
+# Primitive Types
+class Integer(BaseType):
+	openapi_schema_validator = '''{
+		"type": "integer"
+	}'''
+
+class String(BaseType):
+	openapi_schema_validator = '''{
+		"type": "string"
+	}'''
+
+class Float(BaseType):
+	openapi_schema_validator = '''{
+		"type": "number"
+	}'''
+
+class Bool(BaseType):
+	openapi_schema_validator = '''{
+		"type": "boolean"
+	}'''
+
+class List(BaseType):
+	openapi_schema_validator = '''{
+		"type": "array"
+	}'''
+
+class Dict(BaseType):
+	openapi_schema_validator = '''{
+		"type": "object",
+	}'''
+
+# GCP Types
+class GCSPath(BaseType):
+	openapi_schema_validator = '''{
+		"type": "string",
+		"pattern": "^gs://$"
+	}'''
+
+	def __init__(self, path_type='', file_type=''):
+		'''
+		Args
+		:param path_type: describes the paths, for example, bucket, directory, file, etc
+		:param file_type: describes the files, for example, JSON, CSV, etc.
+		'''
+		self.path_type = path_type
+		self.file_type = file_type
+
+class GCRPath(BaseType):
```

Comment:
This is used for specifying an image location in dsl right? I don't think we need this to be strongly typed. It could be from docker, from a private registry, or from GCR. A string suffices in this case.